### PR TITLE
stop newlines being eaten in XML parsing of CDATA

### PR DESCRIPTION
**Pull request by ianturton**
Initial fix to prevent GeoTools eating the newlines in CDATA blocks of WPS requests. Needs two minor fixes to existing tests as I can't limit the behaviour to just CDATA blocks as that information seems to be lost by the time GT currently collapses the white space.

### CODE REVIEW

**Comment by aaime** [comment]
Would be nice to have a test that checks CDATA though ;)

**Comment by ianturton** [comment]
I just realised that the test in the GS commit - I'll move it to GT tomorrow (or later this evening)

File: modules/extension/xsd/xsd-core/src/test/java/org/geotools/xml/impl/ParseExecutorTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        Configuration configuration = new MLConfiguration();
+        Parser parser = new Parser(configuration);
+        parser.setValidating(true);
+
+        HashMap<String,String> parsed;
+
+        InputStream reader = new ByteArrayInputStream(data.getBytes());
+        parsed = (HashMap<String, String>) parser.parse(reader );
+        for(Entry<String, String> e:parsed.entrySet()) {
+            System.out.println(e);
```
Comment (aaime) [inline modules/extension/xsd/xsd-core/src/test/java/org/geotools/xml/impl/ParseExecutorTest.java]: Leftover printout

File: modules/extension/xsd/xsd-core/src/test/java/org/geotools/xml/impl/ParseExecutorTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            if(e.getKey()!=null && e.getKey().equals("mimeType")) {
```
Comment (aaime) [inline modules/extension/xsd/xsd-core/src/test/java/org/geotools/xml/impl/ParseExecutorTest.java]: Formatting is not conforming with our guidelines (@dromagno always gets me on these too :-p )

File: modules/extension/xsd/xsd-core/src/main/java/org/geotools/xml/impl/ParseExecutor.java [hunk 3]
```diff
@@ -326,10 +327,13 @@ public XSDNamedComponent getDeclaration() {
         } else {
             //type is not simple, or complex with simple content, do a check 
             // for mixed
-            if (instance.getTypeDefinition() instanceof XSDComplexTypeDefinition
-                    && ((XSDComplexTypeDefinition) instance.getTypeDefinition()).isMixed()) {
-                //collape the text
-                text = Whitespace.COLLAPSE.preparse(text);
+            //TODO: If this came from a CDATA block we should not collapse it,
```
Comment (aaime) [inline modules/extension/xsd/xsd-core/src/main/java/org/geotools/xml/impl/ParseExecutor.java]: This is a TODO, but still, it removes whitespace collapsing which is normally required (outside of CDATA)

Comment (aaime) [inline modules/extension/xsd/xsd-core/src/main/java/org/geotools/xml/impl/ParseExecutor.java]: And still, this is the fix right?

**Comment by ianturton** [comment]
seems as if part of the issue was curl eating new lines if you send data as ASCII - may be other issues but I'm starting again from scratch on this!

**Pull request closed without merging**